src/hub/vendor/customer.py

```diff
@@ -257,159 +256,224 @@ def run(self) -> None:
-        else:
-            logger.error(
-                "customer subscription updated - no userid",
-                error=self.payload.object.customer,
-            )
-            raise ClientError(
-                f"userid is None for customer {self.payload.object.customer}"
-            )
+
+        if user_id is None:
+            logger.error("customer subscription updated - no userid", error=customer_id)
+            raise ClientError(f"userid is None for customer {customer_id}")
+
+        return user_id
 
     def get_latest_invoice(self, invoice_id) -> Optional[Dict[str, Any]]:
+        """
+        Fetch the Invoice object from Stripe by ID.
+        If the charge has not been completed, wait and attempt again to fetch the object.
+        If after 2 attempts, the Invoice has still not been completed :return None
+        :param invoice_id:
+        :return:
+        """
         attempts = 2
-        latest_invoice = stripe.Invoice.retrieve(id=invoice_id)
-        logger.info("get latest invoice", latest_invoice=latest_invoice)
         while attempts > 0:
-            attempts -= 1
-            if latest_invoice.charge is None or latest_invoice.number is None:
-                time.sleep(0.2)
-                self.get_latest_invoice(invoice_id)
-            else:
+            latest_invoice = stripe.Invoice.retrieve(id=invoice_id)
+            logger.info("get latest invoice", latest_invoice=latest_invoice)
+
+            if latest_invoice.charge is not None and latest_invoice.number is not None:
                 return latest_invoice
+
+            attempts -= 1
+            time.sleep(0.2)
+
         return None
 
-    def get_latest_charge(self, charge) -> Optional[Dict[str, Any]]:
+    def get_latest_charge(self, charge_id) -> Optional[Dict[str, Any]]:
+        """
+        Fetch the Charge object from Stripe by ID.
+        If the charge has not been completed, wait and attempt again to fetch the object.
+        If after 2 attempts, the Charge has still not been completed :return None
+        :param charge_id:
+        :return:
+        """
         attempts = 2
-        latest_charge = stripe.Charge.retrieve(id=charge)
-        logger.info("get latest charge", latest_charge=latest_charge)
         while attempts > 0:
-            attempts -= 1
+            latest_charge = stripe.Charge.retrieve(id=charge_id)
+            logger.info("get latest charge", latest_charge=latest_charge)
+
             if (
-                latest_charge.payment_method_details.card.last4 is None
-                or latest_charge.payment_method_details.card.brand is None
+                latest_charge.payment_method_details.card.last4 is not None
+                or latest_charge.payment_method_details.card.brand is not None
             ):
-                time.sleep(0.2)
-                self.get_latest_charge(charge)
-            else:
                 return latest_charge
+
+            attempts -= 1
+            time.sleep(0.2)
```

Comment:
Same as above, should use shared/vendor - retrieve_stripe_charge ln 346

Reply:
removed method and updated references to use shared/vendor method